### ISSUE TITLE
Make a "widget" version + Compress code further.

### ISSUE THREAD

**Issue by PySimpleGUI** [zargogo/LiveTSX#2]
Someone on Reddit mentioned turning it into a desktop widget.  You can easily do this using this by turning off the title bar.  This leaves the user not able to move the window.  So, turn on the `grab_anywhere` parameter for your Window.  You also need a way to exit the program.  You can add  button or some other mechanism.  In this case I knew there's a little red button that's part of the PySimpleGUI code.  I added it as an image and made it clickable.  If you click the little red X, it will close the window.

![image](https://user-images.githubusercontent.com/46163555/82364753-b31bff00-99dd-11ea-9da4-52ecc4dba325.png)

----------------------

Code compression.

When you find yourself copying and pasting the same code, it's a good time to think about a loop, right?

The lengthy list of if statements can be condensed down to a single if statement if you put it in a loop.

```python
    for i in range(1,7):
        if values[f'i{i}'] is not "":
            window[f'o{i}'].update(parsePrice(values[f'i{i}']))
```

-----------------------

The compete program looksl like this:

```python
import PySimpleGUI as sg
import time
import bs4
import requests


def parsePrice(sym):
    r = requests.get("https://web.tmxmoney.com/quote.php?qm_symbol=" + sym)  # Canadian stocks source (TSX)
    soup = bs4.BeautifulSoup(r.text, "html.parser")
    try:
        price = soup.find("div", {"class": "labs-symbol"}).find("span", {"class": "price"}).find("span").text
        return price
    except:
        return "Cannot find"

sg.theme('DarkBrown1')

def StockInfo(line_num):
    return [sg.Input(key=f"i{line_num}", font=('Helvetica', 20), size=(10, 0)), sg.Text("0.00", size=(6, 0), font=('Helvetica', 20), key=f"o{line_num}")]

layout =  [StockInfo(i) for i in range(1, 7)]
layout += [[sg.Image(data=sg.red_x, enable_events=True, key='-EXIT-')]]

window = sg.Window("TSX Stocks", layout, no_titlebar=True, grab_anywhere=True)

while True:  # Event Loop
    event, values = window.read(timeout=800)
    if event in (sg.WIN_CLOSED, '-EXIT-'):  # if user closed the window using X or clicked Quit button
        break

    for i in range(1,7):
        if values[f'i{i}'] is not "":
            window[f'o{i}'].update(parsePrice(values[f'i{i}']))

window.close()
```

**Comment by zargogo** [comment]
Thank you for the awesome tips! Unfortunately, I don't know why, but adding no_titlebar just makes the app invisible ... (?)

**Comment by PySimpleGUI** [comment]
Shouldn't but maybe does something really weird on the Mac.  Depends perhaps on where you installed Python from. If from Homebrew, it's got lots of problems.  If from  Python.org, then generally works better.  Still, not aware of no_titlebar making the window disappear.  Remember, it will not show up on your taskbar.   But it will still be on the screen.  Note the screenshot.

**Comment by zargogo** [comment]
It's really weird. I just tested it again and it just doesn't create a window. I did install it from Homebrew so maybe it's a bug due to that

**Comment by PySimpleGUI** [comment]
Yep..... Homebrew is not your friend when it comes to tkinter.  It's loaded with bugs that are many years old.

My Mac users that switched to using Python releases from Python.org had their problems all suddenly and completely disappear.

You're welcome to search through the PySimpleGUI issues to see if other Mac users have had this problem.  http://Issues.PySimpleGUI.org, but I'm not aware of any.

There are some funny problems on the Raspberry Pi when combining keep on top with no titlebar.

I would debug this like I debug everything.... **_Simplify simplify simplify_**.  Start with no options on your Window creation.  Then add no_titlebar on its own with no other options to see if that causes the problem or if it requires a combination.

But I would for sure upgrade to the python.org release of Python and stop using the homebrew version.

**Comment by zargogo** [comment]
Ima be honest, it's too much work for me right now ;-) my intention was to just create a small app. But if anybody else wants to branch my project and expand the app, then I pass them the torch

**Comment by PySimpleGUI** [comment]
It's OK... I Get it.  It's why it's a "suggestion".  I've got 500 Issues open on my project. I get there not being enough time.  I'm not at all being critical. Want to make sure you understand that I'm impressed with what you've done already and it's great the way it is.  My suggestions were meant as a roadmap ahead if you care to or have the time to take. it.  I hope it didn't come across badly.

**Comment by zargogo** [comment]
Awh noo don’t worry :) I really appreciate your support and that you’re taking the time to help me out! 🤜🤛☺️
